### ISSUE TITLE
ValueError: Overlapping IntervalIndex is not accepted.

### ISSUE THREAD

**Issue by yuanjianwen-gif** [ulelab/peka#6]
I downlowd xl and peak file from https://imaps.goodwright.com/collections/868/ and run: peka -i tardbp-egfp-hd-hek293-1-20201021-ju_mapped_to_genome_single.bed1 -x tardbp-egfp-hd-hek293-1-20201021-ju_mapped_to_genome_single_peaks.bed1 -g $genome -gi $gfai -r $segment -k 6
But got error:
Namespace(alloutputs=False, clusters=5, distalwindow=150, genomefasta='/mnt/1/genome/hg38/hg38.fa', genomeindex='/mnt/1/genome/hg38/hg38.fa.fai', inputpeaks='tardbp-egfp-hd-hek293-1-20201021-ju_mapped_to_genome_single.bed1', inputxlsites='tardbp-egfp-hd-hek293-1-20201021-ju_mapped_to_genome_single_peaks.bed1', kmerlength=6, outputpath='/mnt/9/yuan_jianwen/scTRIBE_project/09.fastaMotif/00.rmPseudo/test', percentile=0.7, regions='/mnt/12/yuan_jianwen/hg38/Homo_sapiens.GRCh38.103.segment.gtf.gz', repeats='unmasked', smoothing=6, specificregion=None, subsample=True, topn=20, window=25)
Getting thresholded crosslinks
Thresholding intron
lenght of df_reg for intron is: 70328
Traceback (most recent call last):
  File "/home/yuan_jianwen/anaconda3/envs/peka/bin/peka", line 8, in <module>
    sys.exit(cli())
  File "/home/yuan_jianwen/anaconda3/envs/peka/bin/peka.py", line 1317, in cli
    subsample,
  File "/home/yuan_jianwen/anaconda3/envs/peka/bin/peka.py", line 1002, in run
    df_txn = get_threshold_sites(sites_file, percentile=percentile)
  File "/home/yuan_jianwen/anaconda3/envs/peka/bin/peka.py", line 445, in get_threshold_sites
    df_cut = cut_sites_with_region(df_reg, df_region)
  File "/home/yuan_jianwen/anaconda3/envs/peka/bin/peka.py", line 376, in cut_sites_with_region
    df_temp = cut_per_chrom(chrom, df_p, df_m, df_region_p, df_region_m)
  File "/home/yuan_jianwen/anaconda3/envs/peka/bin/peka.py", line 363, in cut_per_chrom
    df_xl_p["cut"] = pd.cut(df_xl_p["start"], interval_index_p)
  File "/home/yuan_jianwen/anaconda3/envs/peka/lib/python3.7/site-packages/pandas/core/reshape/tile.py", line 226, in cut
    raise ValueError('Overlapping IntervalIndex is not accepted.')
ValueError: Overlapping IntervalIndex is not accepted.

**Comment by kkuret** [comment]
Hello! You inputed crosslinks under peak file (-i) and peaks under crosslink file (-xl). Try to reverse them and let me know if it resolves your issue.

**Comment by kkuret** [comment]
Also, please use .bed as a file extension, not .bed1

**Comment by yuanjianwen-gif** [comment]
> Hello! You inputed crosslinks under peak file (-i) and peaks under crosslink file (-x). Try to reverse them and let me know if it resolves your issue.

Appreciate it very much! And  sorry to reply you so late becuase of being busy with tough things. I tested newest version of peka with test data and it works fine. Thanks again.
    In my cases, I only have crosslinks-like bed files but not peaks and  do not want to define threshold.  For the purpose, can I use crosslinks-like bed files as peaks files too and set "--percentile" to 0?

**Comment by kkuret** [comment]
Glad to hear it works for you now! 
I don't completely understand the approach you want to take. Do you want to:
a) give the same crosslink file for as -x and as peaks (-i)? This will not work, as PEKA needs "background crosslinks" to calculate motif enrichment. PEKA  defines background as crosslinks that are located outside peaks. Thus, if you provide the same file for crosslinks and peaks, PEKA won't find any background crosslinks and it will fail.
b) give PEKA two separate crosslink files (one that you'd like to use as foreground and one you'd like as background)? If you have this situation, you should combine both foreground and background crosslinks into one file and feed that as inputxlsites (-x), and then feed the foreground crosslinks as inputpeaks (-i) and set percentile threshold to 0. That way PEKA will use the crosslinks you want as foreground and they will be compared to your background crosslinks.

So in short, yes you can use percentile=0 and define your foreground with crosslink-like bed files, by feeding these as peaks. Just make sure there are also background crosslinks present in the inputxlsites (-x).

**Comment by yuanjianwen-gif** [comment]
> 

You are so nice to reply!

As what you mentioned , I did get some results though it was not that good.  But I realized my data is not from CLIP experiment and the distance between peaks and motif  would be much longer( For instance, there is classical motif enriched in 100bp from the peak  using HOMER).  Maybe I should find another way out.

 Anyway, thanks you very much!
![peka_test](https://user-images.githubusercontent.com/75725108/155956796-1ff6f9dc-3187-4bef-83ed-9e42b5c812eb.JPG)

**Comment by kkuret** [comment]
No trouble! 
But just to let you know with regards to my previous reply that I fixed some unexpected behaviour when percentile=0. Before setting percentile to 0 still resulted in filtering out crosslinks with a score of 1, because 1 is the lowest possible score for a crosslink to exist. Now, if percentile==0, PEKA keeps all crosslinks within peaks as foreground.
For the large distance - you can try running PEKA again (make sure to pull the latest version), but increase the window parameter to ~150 and the distal window to 300 to increase the windows that are used to calculate k-mer enrichment. If you do this, the time of analysis will be longer.
Also, if you decide to change the window, the x-axis on plots currently doesn't change proportionally (it is fixed from -50 to  50, but I might change this soon), so if you'd want to look at the distributions of motifs further away, you'd need to re-plot k-mers from the .tsv files. If you are only interested in enrichment of top-kmers, you can just look at PEKA-scores in the output .tsv.
